tests/Main.hs

```diff
@@ -158,24 +158,6 @@ runClashTest = defaultMain $ clashTestRoot
     , clashTestGroup "crc32"
         [ runTest "CRC32" def
         ]
-    , clashTestGroup "i2c"
```

Comment:
The I2C test is:
1. The only test in the clash testsuite that has multiple synthesize annotations
2. Tests `SimIO` in a common user scenario.

So I would like it if this test wasn't removed. And given that the I2C core in clash cores has a new API, perhaps we can either just:
1. Keep the old domain-monomorphic code in the `examples` directory
2. Or move the old domain-monomorphic code to a test directory
So that we still have a test with multiple Synthesize annotations.

Reply:
2. Sounds like the best option to me!

Reply:
Also agree with 2, we can just put the old code in a test directory.